pyfarm/master/api/jobgroups.py

```diff
@@ -0,0 +1,355 @@
+
+        jobtype_name = g.json.pop("main_jobtype")
+        jobtype = JobType.query.filter_by(name=jobtype_name).first()
+        if not jobtype:
+            return (jsonify(
+                error="Jobtype %s not found" % jobtype_name), NOT_FOUND)
+
+        jobgroup = JobGroup(user=user, main_jobtype=jobtype, **g.json)
+        db.session.add(jobgroup)
+        db.session.commit()
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
+        del jobgroup_data["main_jobtype_id"]
+        logger.info("Created job group %s: %r", jobgroup.title, jobgroup_data)
+
+        return jsonify(jobgroup_data), CREATED
+
+    def get(self):
+        """
+        A ``GET`` to this endpoint will return a list of known job groups.
+
+        .. http:get:: /api/v1/jobgroups/ HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                GET /api/v1/jobgroups/ HTTP/1.1
+                Accept: application/json
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 200 OK
+                Content-Type: application/json
+
+                [
+                    {
+                        "id": 2,
+                        "user": "testuser",
+                        "main_jobtype": "Test JobType",
+                        "title": "Test Group"
+                    }
+                ]
+
+        :statuscode 200: no error
+        """
+        out = []
+        for jobgroup in JobGroup.query:
+            jobgroup_data = jobgroup.to_dict(
+                unpack_relationships=["user", "main_jobtype"])
+            del jobgroup_data["user_id"]
+            del jobgroup_data["main_jobtype_id"]
+            out.append(jobgroup_data)
+
+        return jsonify(out), OK
+
+
+class SingleJobGroupAPI(MethodView):
+    def get(self, group_id):
+        """
+        A ``GET`` to this endpoint will return the requested job group
+
+        .. http:get:: /api/v1/jobgroups/<int:id> HTTP/1.1
+
+            **Request**
+
+            .. sourcecode:: http
+
+                GET /api/v1/jobgroups/2 HTTP/1.1
+                Accept: application/json
+
+            **Response**
+
+            .. sourcecode:: http
+
+                HTTP/1.1 200 OK
+                Content-Type: application/json
+
+                {
+                    "id": 2,
+                    "user": "testuser",
+                    "main_jobtype": "Test JobType",
+                    "jobs": [],
+                    "title": "Test Group"
+                }
+
+        :statuscode 200: no error
+        :statuscode 404: the requested job group was not found
+        """
+        jobgroup = JobGroup.query.filter_by(id=group_id).first()
+
+        if not jobgroup:
+            return (jsonify(error="Requested job group %s not found" % group_id),
+                    NOT_FOUND)
+
+        jobgroup_data = jobgroup.to_dict()
+        del jobgroup_data["user_id"]
```

Comment:
Same function but won't cause the whole request to fail if `jobgroup_data` does not contain the keys we're trying to remove:

```python
jobgroup_data.pop("user_id', None)
jobgroup_data.pop("main_jobtype_id", None)
```